app/services/TradingPremisesService.scala

```diff
@@ -17,30 +17,51 @@
 package services
 
 import javax.inject.Singleton
-
 import models.businessmatching.{BusinessActivity, MoneyServiceBusiness}
-import models.tradingpremises.{TradingPremises, WhatDoesYourBusinessDo}
+import models.tradingpremises
+import models.tradingpremises.{TradingPremisesMsbServices, TradingPremises, WhatDoesYourBusinessDo}
+
+
 
 @Singleton
 class TradingPremisesService {
 
 
-  def addBusinessActivtiesToTradingPremises(
-                                             indices: Seq[Int],
-                                             tradingPremises: Seq[TradingPremises],
-                                             activity: BusinessActivity,
-                                             remove: Boolean): Seq[TradingPremises] =
-    patchTradingPremisesBusinessActivities(tradingPremises){ (wdybd, index) =>
-      wdybd.copy({
-        if (indices contains index) {
-          wdybd.activities + activity
-        } else if (remove) {
-          wdybd.activities - activity
-        } else {
-          wdybd.activities
-        }
-      })
+  def updateTradingPremises(
+                             indices: Seq[Int],
+                             tradingPremises: Seq[models.tradingpremises.TradingPremises],
+                             activity: models.businessmatching.BusinessActivity,
+                             msbServicesInput: Option[models.businessmatching.BusinessMatchingMsbServices],
```

Comment:
Can we shorten/remove the package names?

Reply:
Done